### ISSUE TITLE
stats command shows that the subscription is enabled, but no LSNs for some time

### ISSUE THREAD

**Issue by joetynan** [shayonj/pg_easy_replicate#201]
So, something possibly odd has happened and I'm not sure why.  Usually I see multiple slots on my subscription server that appear to be created by the subscription created with pg_easy_replicate, and they often show a [null] value. However, there's one slot that is actualy named for the subscription that is in 'streaming' mode and DOES show changing LSNs.  I am not sure why, but this stream slot has disappeared.  I tried executing a postgres command to disable and re-enable the subscription (I notice pg_easy_replicate's stop_sync actually completely deletes the publisher and subscriber configurations), and that didn't seem to resolve the issue.  

I see that the slots are all executing COPY commands, but it doesn't seem like any data is actually travelling over. I did bump up my max_wal_senders on the target from the default of 4 to 10, increased my max_logical_replication_slots on the target to 20, and max_logical_subscriber_workers to 10, also on the target.  While this seems to have increased the number of slots that pg_easy_replicate is returning, I still don't see any LSNs on them.

**Comment by joetynan** [comment]
Just FYI, basically stopping/starting sync resolves the issue, but that's not entirely ideal since I don't know if pg_easy_replicate will resume replication from where it stopped, or if it will start over from scratch.

**Comment by shayonj** [comment]
Thats odd, going to block some time to look into this proper in the next week and get back.

**Comment by joetynan** [comment]
Well dropping and creating the publisher/subscriber "fixed" it, but still no idea why the streaming slot would just plain disappear.  One thing that might be a reason is that someone else edited the publication settings to remove a couple tables from it, but didn't refresh the subscriber side.  Even with refreshing it, it still didn't resume. So, it could be that however pg_easy_replicate actually creates the publication and subscription, any settings viewed by editing the publication or subscription should not be changed via SQL command.

**Comment by shayonj** [comment]
> pg_easy_replicate will resume replication from where it stopped, 

No, pg_easy_replicate, which uses subscription/publication under the hood from Postgres, won't resume from where it left off. It will try to perform an initial copy first and then replicate changes. So in case of issues, it's best to start over. It would be nice to have a resume functionality, which is a bit tricky to do inside pg_easy_replicate, but is something that [pg_flo](https://github.com/pgflo/pg_flo) supports.

That said - if there were no LSNs, I have typically found something going wrong with the publication creation state or that the initial copy itself could be failing on the target data, and usually Postgres logs would have more info. Potential reasons could be:

- Inserting data is running into potential constraints and failing
- Replication workers or the copy worker unexpectedly died
- Connection issues between the source <> target database during the replication

In the last two cases, starting again could typically fix the issue, so I think that's likely what you're seeing.

> So, it could be that however pg_easy_replicate actually creates the publication and subscription, any settings viewed by editing the publication or subscription should not be changed via SQL command.

Yes, that is correct - it's better to manage it via pg_easy_replicate. It supports `--exclude-tables` FWIW. If you also update/stop/remove the publication outside of pg_easy_replicate, there could be partial state left over in the `groups` table, so that could cause issues for the same `group-name` in the future, when starting bootstrap, start_sync, or something else.

If you have any logs from start_sync or other commands, I'm happy to help debug.
